flake.nix

```diff
@@ -0,0 +1,100 @@
+          ./integrations/categories/integration { };
+        "categorifier-categories-integration-test" =
+          self.callCabal2nix "categorifier-categories-integration-test"
+          ./integrations/categories/integration-test { };
+        "categorifier-concat-examples" =
+          self.callCabal2nix "categorifier-concat-examples"
+          ./integrations/concat/examples { };
+        "categorifier-concat-integration" =
+          self.callCabal2nix "categorifier-concat-integration"
+          ./integrations/concat/integration { };
+        "categorifier-concat-integration-test" =
+          self.callCabal2nix "categorifier-concat-integration-test"
+          ./integrations/concat/integration-test { };
+        "categorifier-concat-extensions-category" =
+          self.callCabal2nix "categorifier-concat-extensions-category"
+          ./integrations/concat-extensions/category { };
+        "categorifier-concat-extensions-integration" =
+          self.callCabal2nix "categorifier-concat-extensions-integration"
+          ./integrations/concat-extensions/integration { };
+        "categorifier-concat-extensions-integration-test" =
+          self.callCabal2nix "categorifier-concat-extensions-integration-test"
+          ./integrations/concat-extensions/integration-test { };
+        "categorifier-unconcat-category" =
+          self.callCabal2nix "categorifier-unconcat-category"
+          ./integrations/unconcat/category { };
+        "categorifier-unconcat-integration" =
+          self.callCabal2nix "categorifier-unconcat-integration"
+          ./integrations/unconcat/integration { };
+        "categorifier-unconcat-integration-test" =
+          self.callCabal2nix "categorifier-unconcat-integration-test"
+          ./integrations/unconcat/integration-test { };
+        "categorifier-vec-integration" =
+          self.callCabal2nix "categorifier-vec-integration"
+          ./integrations/vec/integration { };
+        "categorifier-vec-integration-test" =
+          self.callCabal2nix "categorifier-vec-integration-test"
+          ./integrations/vec/integration-test { };
+        "categorifier-plugin" =
+          self.callCabal2nix "categorifier-plugin" ./plugin { };
+        "categorifier-plugin-test" =
+          self.callCabal2nix "categorifier-plugin-test" ./plugin-test { };
+        "categorifier-th" = self.callCabal2nix "categorifier-th" ./th { };
+
+      };
+
+    in {
+      # see these issues and discussions:
+      # - https://github.com/NixOS/nixpkgs/issues/16394
+      # - https://github.com/NixOS/nixpkgs/issues/25887
+      # - https://github.com/NixOS/nixpkgs/issues/26561
+      # - https://discourse.nixos.org/t/nix-haskell-development-2020/6170
+      overlay = final: prev: {
+        haskellPackages = prev.haskellPackages.override (old: {
+          overrides = final.lib.composeExtensions (old.overrides or (_: _: { }))
+            haskellOverlay;
+        });
+      };
+
+      devShell.x86_64-linux = let
```

Comment:
Here also, shouldn't be linux-specific.

Reply:
will update with `flake-utils.eachSystem`, too.